packages/insomnia-app/app/models/helpers/__tests__/settings.test.ts

```diff
@@ -0,0 +1,208 @@
+    getConfigSettings.mockReturnValue({});
+    const settings = models.settings.init();
+
+    const result = omitControlledSettings(settings)({ disablePaidFeatureAds: true });
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it('omits settings controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      incognitoMode: true,
+    };
+    const result = omitControlledSettings(settings)({ enableAnalytics: true });
+
+    expect(result).toMatchObject({});
+  });
+
+  it('does not omit settings not controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings = models.settings.init();
+    const result = omitControlledSettings(settings)({ disablePaidFeatureAds: true });
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+});
+
+describe('getControlledSettings', () => {
+  it('overwrites config controlled settings', () => {
+    getConfigSettings.mockReturnValue({ disablePaidFeatureAds: true });
+    const settings: Settings = {
+      ...models.settings.init(),
+      disablePaidFeatureAds: false,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it('does not overwrite settings not controlled by the config', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      disablePaidFeatureAds: true,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it('overwrites settings controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      incognitoMode: true,
+      enableAnalytics: true,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ enableAnalytics: false });
+  });
+
+  it('does not overwrite settings not controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      disablePaidFeatureAds: true,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it.skip('config control trumps (non-config) settings control', () => {
+    getConfigSettings.mockReturnValue({ enableAnalytics: true });
+    const settings: Settings = {
+      ...models.settings.init(),
+      incognitoMode: true,
+      enableAnalytics: false,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ enableAnalytics: true });
+  });
+
+  it('config control gets priority over simple settings control', () => {
```

Comment:
```suggestion
  it('prioritizes config control over simple settings control', () => {
```

This test is also identical to the one above...

Reply:
thanks, updated the copy in 84ec07d4e3b2d1acba73d4b7240216c0441ce9f8

Reply:
see https://github.com/Kong/insomnia/pull/4031#discussion_r726085555 (it's removed now)